src/BinomialQueue/Internals.hs

```diff
@@ -219,11 +228,24 @@ unions = foldl' union empty
 
 -- | \(O(n)\). Map elements and collect the 'Just' results.
 mapMaybe :: Ord b => (a -> Maybe b) -> MinQueue a -> MinQueue b
-mapMaybe f (MinQueue ts) = mapMaybeQueue f (const empty) empty ts
+mapMaybe f = flip foldlU' empty $ \q a ->
+  case f a of
+    Nothing -> q
+    Just b -> insertEager b q
+-- This seems to be needed for specialization.
```

Comment:
Also see #94, most functions don't seem to be specialized.

Reply:
Yeah, we need to deal with that. One step at a time ....